app-shells/scsh/scsh-0.7_pre20160820.ebuild

```diff
@@ -0,0 +1,63 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+# compatible scheme48 version
+SCHEME48V="1.9.2"
+
+inherit autotools
+
+SCSH_HASH="114432435e4eadd54334df6b37fcae505079b49f"
+RX_HASH="d3231ad13de2b44e3ee173b1c9d09ff165e8b6d5"
+
+DESCRIPTION="Unix shell embedded in Scheme (image)"
+HOMEPAGE="https://www.scsh.net/"
+SRC_URI="
+	https://github.com/scheme/scsh/archive/${SCSH_HASH}.tar.gz -> scsh-${PV}.tar.gz
+	https://github.com/scheme/rx/archive/${RX_HASH}.tar.gz -> scsh-${PV}_rx.tar.gz
+"
+S="${WORKDIR}/scsh-${SCSH_HASH}"
+
+RESTRICT="test"
+LICENSE="BSD"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+
+# older versions were in conflict with scheme48's files, 2016 version uses them
+DEPEND="
+	~dev-scheme/scheme48-${SCHEME48V}
+	~app-shells/scsh-lib-${PV}
+"
+RDEPEND="${DEPEND}"
+
+src_unpack() {
+	unpack ${P}.tar.gz
+
+	tar xf "${DISTDIR}/${P}_rx.tar.gz" --strip-components 1 -C "${S}/rx/" ||
+		die "Failed to unpack ${P}_rx.tar.gz"
+}
+
+src_prepare() {
+	SCHEME48VM=/usr/$(get_libdir)/scheme48-${SCHEME48V}/scheme48vm
+	export SCHEME48VM
+
+	default
+	eautoreconf
+}
+
+src_configure() {
+	econf --with-scheme48=${SCHEME48VM}
+}
+
+src_compile() {
+	emake SCHEME48VM=${SCHEME48VM} scsh.image
+}
+
+src_install() {
+	emake DESTDIR="${T}/install" SCHEME48VM=${SCHEME48VM} enough dirs install-scsh
```

Comment:
because we need `install` from Makefile to correct the image's load paths. It has to install to some path from which we then extract `scsh.image`.

Reply:
That's just upstream sillyness. If you look at `Makefile.in`:

```Makefile
install-scsh: scsh
...
        for file in $(patsubst c/%,%,$(COMPILED_LIBS)) ; do \
        $(INSTALL) c/$$file $(DESTDIR)$(LIB)/$$file ; \
        done
...
```

it's installing the lib files to `$(DESTDIR)$(LIB)`. Now look at the build rule for the image:
```Makefile
install-scsh-image: install-scsh
        $(srcdir)/build/build-image.sh $(srcdir) \
                "$(LIB)/" '$(DESTDIR)$(LIB)/scsh.image' '$(SCHEME48) -h 0' '$(LOADS)'
```

it's not pulling the libs from `$(DESTDIR)`, it's relative to `/` now. So that necessitates the scsh-lib package, which installs these libs first. Take a look again at the prerequisite for the rule, it depends on `install-scsh`. Which is dumb, because it's not consuming the output of `install-scsh`, unless your DESTDIR is `/`. So you can't call `install-scsh-image` without introducing a file collision with `scsh-lib`. Thus, you let it install to a temporary location, and pick out `scsh.image` out of that.

Alternatively, you can sed out the prerequisite with:
```sh
sed -e '/install-scsh-image:/ s|:.*|:|' -i Makefile.in
```
in which case you don't need to use a temporary dir, you can just use `DESTDIR="${D}"`